test/validation/api/event/event.c

```diff
@@ -1,16 +1,557 @@
+		if (odp_pool_destroy(ctx->type[i].pool))
+			ret = -1;
+		ctx->type[i].pool = ODP_POOL_INVALID;
+	}
+
+	return ret;
+}
+
+static void event_test_types_and_subtypes(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	odp_event_type_t exp_type;
+	odp_event_type_t type;
+	odp_event_type_t types_out[EVENT_BURST];
+	odp_event_subtype_t exp_sub;
+	odp_event_subtype_t sub;
+	odp_event_subtype_t subs_out[EVENT_BURST];
+	odp_event_t ev;
+	odp_event_t events[MAX_EVENT_TYPES];
+	int i, j, idx;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		exp_type = ctx->type[i].event_type;
+		exp_sub = (exp_type == ODP_EVENT_PACKET) ? ODP_EVENT_PACKET_BASIC
+							 : ODP_EVENT_NO_SUBTYPE;
+
+		for (j = 0; j < EVENT_BURST; j++) {
+			ev = ctx->type[i].events[j];
+
+			CU_ASSERT(odp_event_type(ev) == exp_type);
+			CU_ASSERT(odp_event_subtype(ev) == exp_sub);
+
+			sub = ODP_EVENT_NO_SUBTYPE;
+			CU_ASSERT(odp_event_types(ev, &sub) == exp_type);
+			CU_ASSERT(sub == exp_sub);
+		}
+
+		odp_event_types_multi(ctx->type[i].events, types_out, subs_out, EVENT_BURST);
+		for (j = 0; j < EVENT_BURST; j++) {
+			CU_ASSERT(types_out[j] == exp_type);
+			CU_ASSERT(subs_out[j] == exp_sub);
+		}
+
+		for (j = 0; j < EVENT_BURST; j++)
+			types_out[j] = ODP_EVENT_ANY;
+
+		odp_event_types_multi(ctx->type[i].events, types_out, NULL, EVENT_BURST);
+		for (j = 0; j < EVENT_BURST; j++)
+			CU_ASSERT(types_out[j] == exp_type);
+
+		CU_ASSERT(odp_event_type_multi(ctx->type[i].events, EVENT_BURST, &type)
+			  == EVENT_BURST);
+		CU_ASSERT(type == exp_type);
+
+		/* Fill array indexes 1, ..., len-1 with another event type */
+		idx = (i + 1) % ctx->num_types;
+		for (j = 1; j < ctx->num_types; j++)
+			events[j] = ctx->type[idx].events[j];
+
+		for (j = 0; j < ctx->num_types; j++) {
+			events[j] = ctx->type[i].events[j];
+			CU_ASSERT(odp_event_type_multi(events, ctx->num_types, &type) == j + 1);
+		}
+	}
+}
+
+static void event_test_pool(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	odp_event_t ev;
+	odp_event_type_t exp_type;
+	odp_pool_t ev_pool;
+	int i, j;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		for (j = 0; j < EVENT_BURST; j++) {
+			ev = ctx->type[i].events[j];
+			exp_type = ctx->type[i].event_type;
+			ev_pool = odp_event_pool(ev);
+
+			if (exp_type == ODP_EVENT_TIMEOUT ||
+			    exp_type == ODP_EVENT_DMA_COMPL ||
+			    exp_type == ODP_EVENT_ML_COMPL)
+				CU_ASSERT(ev_pool == ODP_POOL_INVALID);
+			else
+				CU_ASSERT(ev_pool == ctx->type[i].pool);
+		}
+	}
+}
+
+static void event_test_user_area_flag(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	odp_event_t ev;
+	int i, j, flag;
+	void *user_area;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		for (j = 0; j < EVENT_BURST; j++) {
+			ev = ctx->type[i].events[j];
+			user_area = odp_event_user_area(ev);
+
+			CU_ASSERT(user_area != NULL);
+
+			flag = 0;
+			odp_event_user_flag_set(ev, 1);
+			user_area = odp_event_user_area_and_flag(ev, &flag);
+
+			CU_ASSERT(user_area != NULL);
+			CU_ASSERT(flag != 0);
```

Comment:
Could check that the return value matches API spec: "User flag value pointer for output. >0 if user flag is set, 0 if flags is not set, or <0 if event does not have user flag."